### ISSUE TITLE
Support saving tensors to a file in Model addition debuggers

### ISSUE THREAD

**Issue by RyanMullins** [huggingface/transformers#37983]
### Feature request

Current state: [`model_addition_debugger_context` context manager](https://huggingface.co/docs/transformers/v4.51.3/en/internal/model_debugging_utils#model-addition-debuggers) write out a `FULL_TENSORS.json` file containing `repr()`-ized version of the relevant tensors. 

Request: Add a `use_repr: bool` option to `model_addition_debugger_context` that works exactly as above when `True` but saves the full tensors to disk using SafeTensors when set to `False`.

### Motivation

Some model adders, including me in my role as the Hugging Face lead for Gemma, release models on many platforms and need to perform numerical validations of logits across those implementations to ensure consistency and quality. The `repr()`-ized version in the current version of `model_addition_debugger_context` is insufficient to support these sorts of validations. Dumping the tensors to the JSON file will induce _a lot_ of bloat, so the best and most flexible option is to save the tensors to discrete files and reference those files from the `FULL_TENSORS.json` file. 

### Your contribution

I am working on an implementation of this feature.

**Comment by RyanMullins** [comment]
cc @molbap @ArthurZucker
